src/hotspot/share/classfile/dictionary.hpp

```diff
@@ -147,8 +141,8 @@ class DictionaryEntry : public HashtableEntry<InstanceKlass*, mtClass> {
     return (DictionaryEntry**)HashtableEntry<InstanceKlass*, mtClass>::next_addr();
   }
 
-  ProtectionDomainEntry* pd_set() const            { return _pd_set; }
-  void set_pd_set(ProtectionDomainEntry* new_head) {  _pd_set = new_head; }
+  ProtectionDomainEntry* pd_set() const            { return Atomic::load(&_pd_set); }
+  void set_pd_set(ProtectionDomainEntry* entry)    { Atomic::store(&_pd_set, entry); }
```

Comment:
These probably need to be load_acquire and release_store, so that a lock-free traversal can proceed correctly with a locked addition/removal. With suitable name changes for the methods.

Reply:
I forgot that the method names have to match the implementation details. Thanks I didn't know whether they needed acquire etc.